submitted_models/bosdyn_spot/specifications.md

```diff
@@ -0,0 +1,235 @@
+
+The base-footprint odometry is published on topic `base_to_footprint_pose` of type
+`geometry_msgs/PoseWithCovarianceStamped` and defines the pose of the body `base_link` frame
+relative to the projection of the robot to the (flat) ground. This odometry enters a Kalman filter
+which produces TF frame `base_link->base_footprint` (both prefixed by the robot name). A sample
+configuration of this filter is in `config/ekf/base_to_footprint.yaml`, but the users can use their
+own configuration.
+
+The footprint-odom part is published on topic `odom/raw` of type `nav_msgs/Odometry`. This is the
+transformation between the robot's footprint and the odom frame. This odometry enters a Kalman filter
+which produces TF frame `base_footprint->odom` (both prefixed by the robot name). A sample
+configuration of this filter is in `config/ekf/footprint_to_odom.yaml`, but the users can use their
+own configuration.
+
+## Usage Rights
+No additional restrictions have to be taken into account for this model.
+
+It is based on the BSD-licensed packages [spot_ros](https://github.com/clearpathrobotics/spot_ros)
+and CHAMP's [spot_description](https://github.com/chvmp/spot_ros/tree/gazebo/spot_description).
+
+## Cost and Scale
+* Base vehicle: $75,000
+
+Its weight is approximately 34 kg. 
+
+## Sensors
+The bare body of Spot includes the following sensors. The specifications for these instruments are not
+public, so their properties and performance was estimated from real robot data.
+
+* 5 mono fisheye cameras with resolution 640x480 px modeled by `camera` sensor and `MonoCameraSystem` plugin
+  that applies a simple transformation to the original RGB images the simulator produces.
+* 5 stereo cameras with resolution 424x240 px modeled by `depth_camera` sensor.
+* IMU modeled by the `imu` sensor (not exposed via original bosdyn API, but it has to have some).
+* Foot contact sensors publishing the contact data to topic `leg_contacts`.
+
+Additional sensor configs of Spot are expected to add a lot of payload and thus a separate `specifications.md`
+file is expected to be a part of the sensor config packages.
+
+## Control
+
+Even without the support package, the joints can be controlled separately.
+They offer both a positional and a force/effort control interface.
+
+There are several control topics that repeat for every joint and contain the joint name in their name.
+The robot has 4 legs: `front_left`, `front_right`, `rear_left` and `rear_right`. Each leg has 3 joints:
+`hip_x`, `hip_y` and `knee`. So there are 12 joints in total, and their name follows pattern `$leg_$joint`.
+All of these joints are revolute with limits.
+
+### Single-joint control
+
+To control each joint separately via commanded effort, there are topics `cmd_force/front_left_hip_x` etc.
+of type `std_msgs/Float64`. There might be time sync problems when commanding each joint separately, so
+it is suggested to not used this interface and use the multi-commanding topic described further.
```

Comment:
This paragraph is duplicated in the next paragraph.

Reply:
It's not an exact duplicate. The first one talks about effort-control and the second one about position control. I'll try to reword it so that it looks less confusing.